ivre/active/data.py

```diff
@@ -417,6 +418,17 @@ def merge_scanner_scripts(
         nports = len(ports["ports"])
         res["ports"][proto]["count"] = nports
         res["ports"]["count"] = res["ports"].get("count", 0) + nports
+        # If the list of scanned ports is almost exactly the same as
+        # the list of ports scanned by default by a scanner, assume it
+        # is.
+        for (dports, scanner_name) in DEFAULT_SCANNED_PORTS.get(proto, []):
+            if scanner_name not in res.get("scanners", []):
+                if abs(nports / len(dports) - 1) < 0.03 and set(
```

Comment:
This checks that the set of all scanned ports matches the 1000 nmap ports with an 97% accuracy. I'd say that if this ever happens, the host was indeed using nmap at some point, even if some ports are added later.
I'm not sure I understand the case, but I can of course update it

Reply:
I understand what you mean, and even if I keep thinking it would be better to remove such a label when more ports are added, I don't have a strong argument to support that, so I'm fine with your code as it is.

Thanks for the discussion!